File: git과python/python_basic/lotto_api.js
const axios = require('axios');
const _ = require('lodash');
// axios를 활용하여
// https://www.dhlottery.co.kr/common.do?method=getLottoNumber&drwNo=874
// 로 요청을 보내서

// 1등 번호 6개가 담긴 winner 배열을 만들어 출력하세요
// > [1, 5, 19, 23, 28, 42]
let response;
let arr = [];
// axios는 결과값이 프로미스 객체로 오기때문에 .then을 사용해야한다.
axios.get('https://www.dhlottery.co.kr/common.do?method=getLottoNumber&drwNo=874')
    .then((res)=>{
        response = res.data;
        for(let i=1; i<=6; i++){
            // let name = 'drwtNo'+i;
            arr.push(response[`drwtNo${i}`]);
        }
        console.log(_.sortBy(arr));
    })
    .catch((err)=>{
        console.log('오류 : '+err);
    })
// json 파일은 그저 문자열일 뿐이다. object와 비슷하게 생겼을 뿐.

File: git과python/python_basic/lotto_api.py
import requests
from pprint import pprint as pp
# json 이쁘게 보여줌
url = 'https://www.dhlottery.co.kr/common.do?method=getLottoNumber&drwNo=874'
res = requests.get(url)
# json 자체는 문자열일 뿐이기 때문에 python의 객체인 dictionary로 바꿔줘야 한다(파싱)

# print(pp(res.json()))
data = res.json()
winner = []

# js의 push와 거의 동일
for i in range(1,7):
    # print(data[f'drwtNo{i}'])
    # JavaScript의 push 대신 append
    winner.append(data[f'drwtNo{i}'])
print(winner)